ocaml/xapi/xapi_pool.ml

```diff
@@ -410,7 +410,11 @@ let pre_join_checks ~__context ~rpc ~session_id ~force =
   in
   (* Allow pool-join if host does not have any tunnels *)
   let assert_no_tunnels_on_me () =
-    if Db.Tunnel.get_all ~__context <> [] then (
+    let tunnels =
+      Db.Tunnel.get_refs_where ~__context
+        ~expr:(Not (Eq (Field "protocol", Literal "vxlan_mesh")))
```

Comment:
Why a tunnel for "private network" makes difference here?

Reply:
Yeah, there should no difference here. A cross-server private network is limit in a pool.